client/ayon_core/hosts/traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -0,0 +1,696 @@
+                explicit_output_name = None
+                thumbnails_processed = False
+                for filepath, repre_data in representations.items():
+                    # check if any review derivate tag is present
+                    reviewable = any(
+                        tag for tag in repre_data.get("tags", [])
+                        # tag can be `ftrackreview` or `review`
+                        if "review" in tag
+                    )
+                    # since we need to populate multiple thumbnails as
+                    # representation with outputName for (Ftrack instance
+                    # integrator) pairing with reviewable video representations
+                    if (
+                        thumbnails
+                        and multiple_thumbnails
+                        and reviewable
+                    ):
+                        # multiple unique thumbnails per representation needs
+                        # grouping by outputName
+                        # mainly used in Ftrack instance integrator
+                        explicit_output_name = repre_data["representationName"]
+                        relative_thumbnail_path = repre_data["thumbnailPath"]
+                        # representation might not have thumbnail path
+                        # so ignore this one
+                        if not relative_thumbnail_path:
+                            continue
+                        thumb_dir, thumb_file = \
+                            self._get_refactor_thumbnail_path(
+                                staging_dir, relative_thumbnail_path)
+                        filename, ext = os.path.splitext(thumb_file)
+                        thumbnail_repr_data = deepcopy(
+                            base_thumbnail_repre_data)
+                        thumbnail_repr_data.update({
+                            "name": "thumbnail_{}".format(filename),
+                            "ext": ext[1:],
+                            "files": thumb_file,
+                            "stagingDir": thumb_dir,
+                            "outputName": explicit_output_name,
+                        })
+                        new_instance["prepared_data_for_repres"].append(
+                            ("_thumbnail_", thumbnail_repr_data)
+                        )
+                    elif (
+                        thumbnails
+                        and not multiple_thumbnails
+                        and not thumbnails_processed
+                        or not reviewable
+                    ):
+                        if not thumbnails:
+                            continue
+                        # here we will use only one thumbnail for
+                        # all representations
+                        relative_thumbnail_path = repre_data["thumbnailPath"]
+                        if not relative_thumbnail_path:
+                            relative_thumbnail_path = thumbnails.pop()
```

Comment:
I have added comment above. But, it is popping a last thumbnail from the list and since it is only one and we do not need to iterate again over it we can afford to pop.